frontend/src/components/pages/user/ClientRentalsPage.jsx

```diff
@@ -38,28 +58,49 @@ export function ClientRentalsPage() {
         return date.toLocaleString('en-GB', options);
     }
 
+    function isMoreThan24Hours(startTime) {
+        const currentTime = new Date().getTime();
+        const rentalTime = new Date(startTime).getTime();
+        return (rentalTime - currentTime) > (24 * 60 * 60 * 1000);
+    }
+
     return (
         <div className='client_rentals_page page_content'>
             <h2>Rentals</h2>
             <table>
                 <thead>
-                    <th>Brand</th>
-                    <th>Model</th>
-                    <th>Registration number</th>
-                    <th>Start time</th>
-                    <th>End time</th>
-                    <th>Status</th>
-                    <th>Policy</th>
+                <th>Brand</th>
+                <th>Model</th>
+                <th>Registration number</th>
+                <th>Start time</th>
+                <th>End time</th>
+                <th>Status</th>
+                <th>Policy</th>
+                <th>Actions</th>
                 </thead>
-                <tbody>{rentals?.map(rental => <tr>
-                    <td>{rental.brand}</td>
-                    <td>{rental.model}</td>
-                    <td>{rental.registration_number}</td>
-                    <td>{formatDate(rental.start_time)}</td>
-                    <td>{formatDate(rental.end_time)}</td>
-                    <td>{rental.rental_status.replace(/_/g, ' ')}</td>
-                    <td>{rental.policy_name}</td>
-                </tr>)}</tbody>
+                <tbody>
+                {rentals?.map(rental => {
+                    return (
+                        <tr key={rental.rental_id}>
+                            <td>{rental.brand}</td>
+                            <td>{rental.model}</td>
+                            <td>{rental.registration_number}</td>
+                            <td>{formatDate(rental.start_time)}</td>
+                            <td>{formatDate(rental.end_time)}</td>
+                            <td>{rental.rental_status.replace(/_/g, ' ')}</td>
+                            <td>{rental.policy_name}</td>
+                            {isMoreThan24Hours(rental.start_time) && rental.rental_status !== "canceled" ?
```

Comment:
W tym momencie można zmienić sobie godzinę na kompie i dalej będzie można anulować rezerwację w dowolnym momencie -> trzeba to sprawdzać dodatkowo po stronie serwera